pymilvus/client/stub.py

```diff
@@ -134,17 +134,19 @@ def __init__(self, host=None, port=None, handler="GRPC", pool="SingletonThread",
 
     @check_connect
     def _wait_for_healthy(self, timeout=30, retry=10):
+        _timeout_on_every_retry = self._kw.get("timeout", None)
```

Comment:
@check_connect
    def _wait_for_healthy(self, timeout=3, retry=10):
        with self._connection() as handler:
            while retry > 0:
                is_except = False
                try:
                    status = handler.fake_register_link(timeout)
                    if status.error_code == 0:
                        self._deploy_mode = status.reason
                        return
                except:
                    is_except = True
                finally:
                    sleep_time = 1 if is_except else timeout
                    time.sleep(sleep_time)
                    retry -= 1
            raise Exception("server is not healthy, please try again later")
            
          Can you refer to the logic above?